contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -134,6 +134,9 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
             "Only authenticated addresses in ovmStateManager can call this function"
         );
 
+        // Initialize the execution context.
+        _initContext(_transaction);
+
```

Comment:
Context has to be initialized *before* we call `_checkNeedsNewEpoch` or the whole transaction fails.

Reply:
I also have some `run` order changes in [this PR](https://github.com/ethereum-optimism/contracts-v2/pull/94/files), but let's merge this in and I can update mine.